cpp/src/parquet/encoding-test.cc

```diff
@@ -314,6 +318,185 @@ TEST(TestDictionaryEncoding, CannotDictDecodeBoolean) {
   ASSERT_THROW(MakeDictDecoder<BooleanType>(nullptr), ParquetException);
 }
 
+// ----------------------------------------------------------------------
+// Shared arrow builder decode tests
+class TestDecodeArrow : public ::testing::Test {
+ public:
+  void SetUp() override {
+    InitDataInputs();
+    SetupEncoderDecoder();
+  }
+
+  void InitDataInputs() {
+    // Build a dictionary array and its dense representation for testing building both
+    auto dict_values = ::arrow::ArrayFromJSON(::arrow::binary(), "[\"foo\", \"bar\"]");
+    auto dtype = ::arrow::dictionary(::arrow::int8(), dict_values);
+    auto indices = ::arrow::ArrayFromJSON(::arrow::int8(), "[0, 1, 0, 0]");
+    ASSERT_OK(::arrow::DictionaryArray::FromArrays(dtype, indices, &expected_dict_));
+    expected_array_ =
+        ArrayFromJSON(::arrow::binary(), "[\"foo\", \"bar\", \"foo\", \"foo\"]");
+    num_values_ = static_cast<int>(expected_array_->length());
+
+    // arrow::BinaryType maps to parquet::ByteArray
+    const auto& binary_array = static_cast<const ::arrow::BinaryArray&>(*expected_array_);
+    for (int64_t i = 0; i < binary_array.length(); i++) {
```

Comment:
Very minor nitpick - I think you can use pre-increment (`++i`) instead of post-increment (`i++`) here and in other `for` loops in this diff in order to [avoid the temporary copy that is returned by the post-increment operator](https://en.cppreference.com/w/cpp/language/operator_incdec#Notes), which isn't required in this context. Doing a [quick search through the Arrow codebase](https://github.com/apache/arrow/search?l=C%2B%2B&q=for+%28int+i) seems to suggest that using pre-increment in similar scenarios is fairly common.

Reply:
Good spot, I found this discussed in the [google c++ style guide](https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement).  I'll go for consistency with the rest of this codebase.